contracts/voting-verifier/src/events.rs

```diff
@@ -29,12 +30,32 @@ pub struct PollStarted {
     pub source_gateway_address: String,
     pub confirmation_height: u64,
     pub expires_at: u64,
-    pub messages: Vec<EvmMessage>,
+    pub data: PollData,
     pub participants: Vec<Addr>,
 }
 
+pub enum PollData {
+    Messages(Vec<EvmMessage>),
+    WorkerSet(WorkerSetConfirmation),
+}
+
+fn get_data_attribute(data: PollData) -> (String, String) {
+    match data {
+        PollData::Messages(msgs) => (
+            "messages".into(),
```

Comment:
I don't think using the attribute name to differentiate is the right way to go. I'd much prefer either 1) the poll type has its own attribute or 2) using a different event type. For the actual data, `data` is good.

Reply:
Out of these 2, I prefer 2)

Reply:
I went with 2, lmk what you think.